File: src/main/java/com/nedap/university/go/gocommands/incomingCommandsToServer/MoveCommand.java
package com.nedap.university.go.gocommands.incomingCommandsToServer;

import com.nedap.university.go.gocommands.Command;
import com.nedap.university.go.server.ClientHandler;

/**
 * Created by martijn.slot on 21/02/2017.
 */
public class MoveCommand extends Command {

    private ClientHandler clientHandler;

    public MoveCommand(String[] splitMessage, ClientHandler clientHandler) {
        super();
        this.splitMessage = splitMessage;
        this.clientHandler = clientHandler;

    }

    @Override
    public void execute() {
        switch (clientHandler.getClientStatus()) {
            case INGAME_TURN:
                if (splitMessage.length == 3 && isParsable(splitMessage[1]) && isParsable(splitMessage[2])) {
                    clientHandler.handleMoveCommand(splitMessage);
                } else {
                    clientHandler.writeToClient("WARNING MOVE command syntax not OK. Please repeat. ");
                }
                break;
            default:
                cannotExecute();
                break;
        }
    }

    @Override
    protected void cannotExecute() {
        clientHandler.writeToClient("WARNING Don't MOVE! Wait for your turn like the rest of us. ");
    }

    private boolean isParsable(String input) {
        try {
            Integer.parseInt(input);
        } catch (NumberFormatException e) {
            return false;
        }
        return true;
    }

}

File: src/main/java/com/nedap/university/go/gocommands/clientToServer/MoveCommand.kt
package com.nedap.university.go.gocommands.clientToServer

import com.nedap.university.go.gocommands.Command
import com.nedap.university.go.server.ClientHandler

/**
 * Created by martijn.slot on 21/02/2017.
 */
class MoveCommand(splitMessage: Array<String>, private val clientHandler: ClientHandler) : Command() {

    init {
        this.splitMessage = splitMessage

    }

    @Override
    fun execute() {
        when (clientHandler.getClientStatus()) {
            INGAME_TURN -> if (splitMessage.length === 3 && isParsable(splitMessage[1]) && isParsable(splitMessage[2])) {
                clientHandler.handleMoveCommand(splitMessage)
            } else {
                clientHandler.writeToClient("WARNING MOVE command syntax not OK. Please repeat. ")
            }
            else -> cannotExecute()
        }
    }

    protected fun cannotExecute() {
        clientHandler.writeToClient("WARNING Don't MOVE! Wait for your turn like the rest of us. ")
    }

    private fun isParsable(input: String): Boolean {
        try {
            Integer.parseInt(input)
        } catch (e: NumberFormatException) {
            return false
        }

        return true
    }

}
